src/ripple/app/ledger/Ledger.cpp

```diff
@@ -416,7 +416,7 @@ Ledger::setAccepted(NetClock::time_point closeTime,
 bool Ledger::addSLE (SLE const& sle)
 {
     SHAMapItem item (sle.getIndex(), sle.getSerializer());
-    return stateMap_->addItem(item, false, false);
+    return stateMap_->addItem(std::move(item), false, false);
```

Comment:
The construction of the `SHAMapItem` costs an allocation and so that's where almost all of the expense is here.  With either version, the number of allocations and deallocations is the same.  So using these guidelines: http://howardhinnant.github.io/coding_guidelines.html that brought it down to readability:  Which is more readable?  I can't get very worked up that one version is more readable than the other.  I'll go with whichever you prefer.